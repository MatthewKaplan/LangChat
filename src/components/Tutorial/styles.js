import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'space-around'
	},
	slideContainer: {
		flex: 1,
		alignItems: 'center',

		height: '100%',
		width: '100%',
		zIndex: 1
	},
	tutorialText: {
		fontSize: 18,
		marginTop: 30,
		marginBottom: 30
	},
	imageStyle: {
		marginTop: 0,
		height: '100%',
		width: '100%',
		opacity: 0.8
	},
	textStyle: {
		alignSelf: 'center',
		color: '#fff',
		fontSize: 19,
		fontWeight: '600',
		paddingTop: 12,
		paddingBottom: 12
	},
	buttonStyle: {
		position: "absolute",
		width: '90%',
		backgroundColor: '#007aff',
		borderRadius: 5,
		alignSelf: 'center',
		top: 650
	}
});
