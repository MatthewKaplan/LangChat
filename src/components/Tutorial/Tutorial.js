import React from 'react';
import { View, Text, StyleSheet, Button, Image } from 'react-native';
import Swiper from 'react-native-web-swiper';
import { Actions } from 'react-native-router-flux';
import { Styles } from './styles';
export default class Tutorial extends React.Component {
	render () {
		return (
			<View style={styles.container}>
				<Swiper>
					<View style={[ styles.slideContainer, styles.slide1 ]}>
						<Image style={styles.imageStyle} source={require('../../../assets/talking.jpg')} />
						<Text>Slide 1</Text>
					</View>
					<View style={[ styles.slideContainer, styles.slide2 ]}>
						<Image style={styles.imageStyle} source={require('../../../assets/learn_with_others.jpg')} />
						<Text>Slide 2</Text>
					</View>
					<View style={[ styles.slideContainer, styles.slide3 ]}>
						<Image style={styles.imageStyle} source={require('../../../assets/group_learning.jpg')} />
						<Text>Slide 3</Text>
					</View>
					<View style={[ styles.slideContainer, styles.slide4 ]}>
						<Button title="Go home" onPress={() => Actions.welcome()} />
					</View>
				</Swiper>
			</View>
		);
	}
}
