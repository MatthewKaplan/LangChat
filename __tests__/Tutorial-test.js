// __tests__/Tutorial-test.js
import React from 'react';
import Tutorial from '../src/components/Tutorial/Tutorial';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
	const tree = renderer.create(<Tutorial />).toJSON();
	expect(tree).toMatchSnapshot();
});
