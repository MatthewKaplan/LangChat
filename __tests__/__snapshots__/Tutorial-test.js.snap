// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorial Component should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</RCTSafeAreaView>
`;
